mkosi/machine.py

```diff
@@ -122,9 +142,46 @@ def run(self, commands: Sequence[str], timeout: int = 900, check: bool = True) -
 
         return process
 
-    def __exit__(self, *args: Any, **kwargs: Any) -> None:
+    def _kill(self, *args: Any, **kwargs: Any) -> None:
         if self._serial:
             self._serial.kill(signal.SIGTERM)
             self.exit_code = self._serial.wait()
             self._serial = None
         self.stack.__exit__(*args, **kwargs)
+
+    def __exit__(self) -> None:
+        self._kill(*exc_info())
+
+    def reboot(self) -> None:
+        self._kill(*exc_info())
+        self._boot()
+
+
+class MkosiMachineTest(unittest.TestCase):
+    mkosi_tests_no_qemu: bool
+    mkosi_tests_no_nspawn: bool
+    args: Sequence[str]
+    machine: Machine
+
+    def __init_subclass__(cls, args: Sequence[str] = []) -> None:
+        cls.args = args
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.machine = Machine(cls.args)
+
+        mkosi_tests_no_qemu = os.environ.get("MKOSI_TESTS_NO_QEMU")
+        mkosi_tests_no_nspawn = os.environ.get("MKOSI_TESTS_NO_NSPAWN")
```

Comment:
Maybe add a check to make sure the values are (1,0) and if anything else, call `die()`?

Reply:
Systemd uses the following logic to parse booleans from environment variables. Let's allow the same strings to be used for mkosi's environment variables.

```

int parse_boolean(const char *v) {
        if (!v)
                return -EINVAL;

        if (STRCASE_IN_SET(v,
                           "1",
                           "yes",
                           "y",
                           "true",
                           "t",
                           "on"))
                return 1;

        if (STRCASE_IN_SET(v,
                           "0",
                           "no",
                           "n",
                           "false",
                           "f",
                           "off"))
                return 0;

        return -EINVAL;
}
```